slatedb/src/retention_iterator.rs

```diff
@@ -84,7 +83,8 @@ impl<T: KeyValueIterator> RetentionIterator<T> {
                     let create_ts = entry
                         .create_ts
                         .expect("a record with no create_ts should not happen");
-                    let current_system_ts = system_time_to_millis(system_clock.now());
+                    // TODO: This is wrong! We're mixing logical (create_ts) and physical (system_clock) timestamps.
```

Comment:
does this need to be addressed in this PR? Should it be an issue so we can track it?

Reply:
I didn't want to address it in this PR since it's a bit unrelated. We're tracking it here: #685